src/textual/widgets/_data_table.py

```diff
@@ -2127,11 +2132,19 @@ def sort_by_column_keys(
             result = itemgetter(*columns)(row_data)
             return result
 
+        _key = key
+        if key and columns:
+
+            def _key(row):
+                return key(itemgetter(*columns)(row[1]))
+
         ordered_rows = sorted(
-            self._data.items(), key=sort_by_column_keys, reverse=reverse
+            self._data.items(),
+            key=_key if _key else sort_by_column_keys,
```

Comment:
Can we have an explicit check for `is not None` here. For the (admittedly unlikely) scenario of a callable object with a `__bool__` that returns False.